app/src/main/java/es/wolfi/passman/API/Credential.java

```diff
@@ -275,6 +335,111 @@ public static Credential fromJSON(JSONObject j, Vault v) throws JSONException {
+            params.put("delete_time", getDeleteTime());
+            params.put("files", files);
+            params.put("custom_fields", customFields);
+            params.put("otp", otp);
+            params.put("compromised", compromised);
+            params.put("hidden", isHidden());
+
+            requestAPI(c, "credentials", params, "POST", responseHandler);
+        } catch (JSONException | MalformedURLException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void update(Context c, final AsyncHttpResponseHandler responseHandler) {
+        RequestParams params = new RequestParams();
+        params.setUseJsonStreamer(true);
+
+        try {
+            JSONObject icon;
+
+            if (favicon == null || favicon.equals("") || favicon.equals("null")) {
+                icon = null;
+            } else {
+                try {
+                    icon = new JSONObject(favicon);
+                } catch (JSONException e) {
+                    e.printStackTrace();
+                    icon = null;
+                }
+            }
+
+            params.put("vault_id", getVaultId());
+            params.put("credential_id", getId());
+            params.put("guid", getGuid());
+            params.put("label", label);
+            params.put("description", description);
+            params.put("created", getCreated());
+            params.put("changed", getChanged());
+            params.put("tags", tags);
+            params.put("email", email);
+            params.put("icon", icon);
+            params.put("username", username);
+            params.put("password", password);
+            params.put("url", url);
+            params.put("renew_interval", getRenewInterval());
+            params.put("expire_time", getExpireTime());
+            params.put("delete_time", getDeleteTime());
+            params.put("files", files);
+            params.put("custom_fields", customFields);
+            params.put("otp", otp);
+            params.put("compromised", compromised);
+            params.put("hidden", isHidden());
+
+            requestAPI(c, "credentials/" + getGuid(), params, "PATCH", responseHandler);
+        } catch (MalformedURLException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void sendFileDeleteRequest(Context c, int file_id, final AsyncHttpResponseHandler responseHandler) {
+        RequestParams params = new RequestParams();
+        try {
+            requestAPI(c, "file/" + file_id, params, "DELETE", responseHandler);
+        } catch (MalformedURLException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void uploadFile(Context c, String encodedFile, String fileName, String mimeType, int fileSize, final AsyncHttpResponseHandler responseHandler) {
+        RequestParams params = new RequestParams();
+        params.setUseJsonStreamer(true);
+
+        try {
+            params.put("filename", vault.encryptString(fileName));
```

Comment:
Isn't this code part of the File class already?

Reply:
No really. Thats about uploading the "real" file.
The created File class contains only the file-"metadata" which is saved in the credential itself.
I think the function is well placed there.
Otherwise we would need a second e.g. RealFile class only for uploading the file. It would not be used at any other time.